src/toil/lib/bioio.py

```diff
@@ -294,27 +294,9 @@ def getTempDirectory(rootDir=None):
     returns a temporary directory that must be manually deleted. rootDir will be
     created if it does not exist.
     """
-    if rootDir is None:
-        return tempfile.mkdtemp()
-    else:
-        if not os.path.exists(rootDir):
-            try:
-                os.makedirs(rootDir)
-            except OSError:
-                # Maybe it got created between the test and the makedirs call?
-                pass
-            
-        while True:
-            # Keep trying names until we find one that doesn't exist. If one
-            # does exist, don't nest inside it, because someone else may be
-            # using it for something.
-            tmpDir = os.path.join(rootDir, "tmp_" + getRandomAlphaNumericString())
-            if not os.path.exists(tmpDir):
-                break
-                
-        os.mkdir(tmpDir)
-        os.chmod(tmpDir, 0777) #Ensure everyone has access to the file.
-        return tmpDir
+    tempdir = tempfile.mkdtemp(dir=rootDir)
+    os.chmod(tmpDir, 0755)
+    return tempdir
```

Comment:
For one, I wouldn't want every temp directory to be chmod-ed.

And without the chmod, getTempDirectory adds nothing to mkdtemp. So getTempDirectory needs to go. You will have to do an exhaustive search for every usage of it. That's what an IDE like PyCharm is for.